src/vtquery.cpp

```diff
@@ -541,44 +476,50 @@ struct Worker : Nan::AsyncWorker {
 
         // number of results to loop through
         v8::Local<v8::Object> results_object = Nan::New<v8::Object>();
-        v8::Local<v8::Array> features_array = Nan::New<v8::Array>(results_queue_.size());
+        v8::Local<v8::Array> features_array = Nan::New<v8::Array>();
         results_object->Set(Nan::New("type").ToLocalChecked(), Nan::New<v8::String>("FeatureCollection").ToLocalChecked());
 
         // for each result object
         while (!results_queue_.empty()) {
             auto const& feature = results_queue_.back(); // get reference to top item in results queue
-            v8::Local<v8::Object> feature_obj = Nan::New<v8::Object>();
-            feature_obj->Set(Nan::New("type").ToLocalChecked(), Nan::New<v8::String>("Feature").ToLocalChecked());
-
-            // create geometry object
-            v8::Local<v8::Object> geometry_obj = Nan::New<v8::Object>();
-            geometry_obj->Set(Nan::New("type").ToLocalChecked(), Nan::New<v8::String>("Point").ToLocalChecked());
-            v8::Local<v8::Array> coordinates_array = Nan::New<v8::Array>(2);
-            coordinates_array->Set(0, Nan::New<v8::Number>(feature->coordinates.x)); // latitude
-            coordinates_array->Set(1, Nan::New<v8::Number>(feature->coordinates.y)); // longitude
-            geometry_obj->Set(Nan::New("coordinates").ToLocalChecked(), coordinates_array);
-            feature_obj->Set(Nan::New("geometry").ToLocalChecked(), geometry_obj);
-
-            // create properties object
-            v8::Local<v8::Object> properties_obj = Nan::New<v8::Object>();
-            for (auto const& prop : feature->properties) {
-                set_property(prop, properties_obj);
-            }
+            // if this is a default value, don't use it
+            if (feature.distance < std::numeric_limits<double>::max()) {
```

Comment:
need to take into account the originally default constructed ResultObjects - they'll add a `null` value to the array